src/v/datalake/record_multiplexer.cc

```diff
@@ -49,31 +68,29 @@ record_multiplexer::record_multiplexer(
   , _table_creator(table_creator)
   , _invalid_record_action(invalid_record_action)
   , _location_provider(std::move(location_provider))
-  , _translation_probe(translation_probe)
-  , _as(as) {}
-
-ss::future<ss::stop_iteration>
-record_multiplexer::operator()(model::record_batch batch) {
-    if (_as.abort_requested()) {
-        vlog(
-          _log.debug,
-          "Abort requested, stopping translation, reason: {}",
-          _as.abort_reason());
-        co_return ss::stop_iteration::yes;
-    }
+  , _translation_probe(translation_probe) {}
+
+ss::future<> record_multiplexer::multiplex(
+  model::record_batch_reader reader,
+  model::timeout_clock::time_point deadline,
+  ss::abort_source& as) {
+    co_await std::move(reader).consume(
+      relaying_consumer{[this, &as](model::record_batch b) mutable {
+          return do_multiplex(std::move(b), as);
+      }},
```

Comment:
huh, it's weird that we didn't already have a `lambda_consumer` or something like that